hypershift-operator/controllers/hostedcluster/internal/platform/azure/azure.go

```diff
@@ -184,14 +193,61 @@ func (a *Azure) ReconcileCredentials(ctx context.Context, c client.Client, creat
 	return nil
 }
 
-func (a *Azure) ReconcileSecretEncryption(ctx context.Context, c client.Client, createOrUpdate upsert.CreateOrUpdateFN, hcluster *hyperv1.HostedCluster, controlPlaneNamespace string) error {
+func (a Azure) ReconcileSecretEncryption(ctx context.Context, c client.Client, createOrUpdate upsert.CreateOrUpdateFN, hcluster *hyperv1.HostedCluster, controlPlaneNamespace string) error {
 	return nil
 }
 
-func (a *Azure) CAPIProviderPolicyRules() []rbacv1.PolicyRule {
+func (a Azure) CAPIProviderPolicyRules() []rbacv1.PolicyRule {
 	return nil
 }
 
-func (a *Azure) DeleteCredentials(ctx context.Context, c client.Client, hcluster *hyperv1.HostedCluster, controlPlaneNamespace string) error {
+func (a Azure) DeleteCredentials(ctx context.Context, c client.Client, hcluster *hyperv1.HostedCluster, controlPlaneNamespace string) error {
+	return nil
+}
+
+func reconcileAzureCluster(azureCluster *capiazure.AzureCluster, hcluster *hyperv1.HostedCluster, apiEndpoint hyperv1.APIEndpoint, controlPlaneNamespace string) error {
+	if azureCluster.Annotations == nil {
+		azureCluster.Annotations = map[string]string{}
+	}
+
+	azureCluster.Annotations[capiv1.ManagedByAnnotation] = "external"
+
+	azureCluster.Spec.Location = hcluster.Spec.Platform.Azure.Location
+	azureCluster.Spec.ResourceGroup = hcluster.Spec.Platform.Azure.ResourceGroupName
+	azureCluster.Spec.NetworkSpec.Vnet.ID = hcluster.Spec.Platform.Azure.VnetID
+	azureCluster.Spec.NetworkSpec.Vnet.Name = hcluster.Spec.Platform.Azure.VnetName
+	azureCluster.Spec.NetworkSpec.Vnet.ResourceGroup = hcluster.Spec.Platform.Azure.ResourceGroupName
+	azureCluster.Spec.SubscriptionID = hcluster.Spec.Platform.Azure.SubscriptionID
+
+	azureCluster.Spec.ControlPlaneEndpoint = capiv1.APIEndpoint{
+		Host: apiEndpoint.Host,
+		Port: apiEndpoint.Port,
+	}
+
+	azureCluster.Status.Ready = true
+
+	azureCluster.Spec.IdentityRef = &corev1.ObjectReference{Name: hcluster.Name, Namespace: controlPlaneNamespace}
+
+	return nil
+}
+
+func reconcileAzureClusterIdentity(ctx context.Context, c client.Client, hcluster *hyperv1.HostedCluster, azureClusterIdentity *capiazure.AzureClusterIdentity, controlPlaneNamespace string) error {
+	var userCloudCredSecret corev1.Secret
+
+	secretName := client.ObjectKey{Namespace: controlPlaneNamespace, Name: hcluster.Spec.Platform.Azure.Credentials.Name}
```

Comment:
why are you fetching the secret here?
If it's a validation check, I think it's fair to assume the secret exist and just propagate the ref.
See `validateAzureConfig`.

Reply:
This was leftover code from a previous implementation I believe. It's been removed.

Reply:
This is needed to get the client and tenant IDs.